multiplex.go

```diff
@@ -214,7 +223,10 @@ func (mp *Multiplex) NewNamedStream(name string) (*Stream, error) {
 	mp.channels[s.id] = s
 	mp.chLock.Unlock()
 
-	err := mp.sendMsg(context.Background(), header, []byte(name))
+	ctx, cancel := context.WithTimeout(bgCtx, time.Minute)
```

Comment:
Can we move these timeouts to a `SocketGracePeriod` var, so they can be adjusted?

Reply:
Sure will do. Do we want separate timeouts for reset and stream open or just use a single one?

Reply:
Done.